02_bubble/nickolee/bubblesort_test.go

```diff
@@ -0,0 +1,40 @@
+package main
+
+import (
+	"bytes"
+	"reflect"
+	"strconv"
+	"testing"
+)
+
+func TestBubbleSortFunc(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	// Test empty slice
+	test1 := bubbleSort([]int{})
+	test1Expected := []int{}
+
+	// Test random slice
+	test2 := bubbleSort([]int{3, 2, 5, 1, 7, 2, 9, 1, 0})
+	test2Expected := []int{0, 1, 1, 2, 2, 3, 5, 7, 9}
+
+	// Check whether expected results equal actual results
+	if !reflect.DeepEqual(test1, test1Expected) || !reflect.DeepEqual(test2, test2Expected) {
+		t.Errorf("TestBubbleSortFunc: Unexpected output in main()")
+	}
```

Comment:
Consider table driver test as referred to in the course https://dave.cheney.net/2019/05/07/prefer-table-driven-tests
Also a wider range of boundary tests like the worst case sorting using reverse numbers 9,8,7, ... 0, and what about negative numbers.

Reply:
Review of colleague's PR noted. I'm happy to approve.

Reply:
I must insist on table driven tests here.

Reply:
Yep have implemented this now